docs/fundamentals/code-analysis/quality-rules/ca1862.md

```diff
@@ -0,0 +1,105 @@
+---
+title: "CA1862: Use the 'StringComparison' method overloads to perform case-insensitive string comparisons"
+description: "Learn about code analysis rule CA1862: Use the 'StringComparison' method overloads to perform case-insensitive string comparisons"
+ms.date: 11/17/2023
+f1_keywords:
+  - "CA1862"
+helpviewer_keywords:
+  - "CA1862"
+dev_langs:
+  - CSharp
+  - VB
+---
+
+# CA1862: Use the 'StringComparison' method overloads to perform case-insensitive string comparisons
+
+| Property                            | Value                                  |
+|-------------------------------------|----------------------------------------|
+| **Rule ID**                         | CA1862                                 |
+| **Title**                           | Use the 'StringComparison' method overloads to perform case-insensitive string comparisons |
+| **Category**                        | [Performance](performance-warnings.md) |
+| **Fix is breaking or non-breaking** | Non-breaking                           |
+| **Enabled by default in .NET 8**    | No                                     |
+
+## Cause
+
+Code compares two strings in a case-insensitive manner by first calling <xref:System.String.ToLower>, <xref:System.String.ToLowerInvariant>, <xref:System.String.ToUpper>, or <xref:System.String.ToUpperInvariant> on one or both strings.
+
+## Rule description
+
+When code calls <xref:System.String.ToLower>, <xref:System.String.ToLowerInvariant>, <xref:System.String.ToUpper>, or <xref:System.String.ToUpperInvariant>, an allocation is performed. If the only reason for calling these methods is to perform a case-insensitive string comparison or search, the allocation is unnecessary. Instead, you can call a string comparison method that takes a <xref:System.StringComparison> and specify one of the `*IgnoreCase` values.
+
+## How to fix violations
+
+Remove the call to <xref:System.String.ToLower>, <xref:System.String.ToLowerInvariant>, <xref:System.String.ToUpper>, or <xref:System.String.ToUpperInvariant>, and either call one of the <xref:System.StringComparer> methods, or one of the following methods that takes a <xref:System.StringComparison> argument:
+
```

Comment:
The CA1862 behavior change was that we won't offer a fix when the user combines cultures (for example: ToUpper vs ToUpperInvariant).

We also updated the diagnostic description to include a warning for the user telling them that they need to test the fix if they decide to apply it, or use Ordinal if needed.